modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/editor_image_uploader.js

```diff
@@ -319,8 +311,7 @@ AUI.add(
 			},
 		});
 
-		A.Plugin.LiferayBlogsUploader = EditorImageUploader;
-		A.Plugin.LiferayEditorImageUploader = EditorImageUploader;
+		Liferay.EditorImageUploader = EditorImageUploader;
```

Comment:
Would be nice if we could find a way to do this without having to leak it via the `Liferay` global. Can we instead pass this in as a dep to via `<aui:script require="...">`?

Reply:
Or even better... why not just bake it into the plugin itself?

As with many other things, I feel we decoupled too much here. `EditorImageUploader` is not really being used anywhere else (and probably won't be ever). 

The reason for the current architecture is because we had the `add_images` plugin as something "agnostic" inside AlloyEditor that literally did nothing but allowed custom image handling on integration. This works great on paper, but produces this situation where we have to bootstrap and update 2 different things at once because they're not really decoupled at all :D 

Since we now have the plugin here, we could simply do the whole thing in there, avoid the event passing (keep it for backwards compatibility, if anything) and simplify the editors bootstrapping process.

Reply:
> Or even better... why not just bake it into the plugin itself?

I did not realize that `alloyeditor` can use `addimages`, instead of `ae_addimages`! Yeah, we can greatly simplify this.